File: Client/Window/angular/projects/main/src/app/components/lobbychoice/models/data-for-custom-lobby-creation.ts
import { CustomLobbyWeaponData } from './custom-lobby-weapon-data';
import { CustomLobbyArmsRaceWeaponData } from './custom-lobby-armsraceweapon-data';

export interface DataForCustomLobbyCreation {
    /** WeaponDatas */
    [0]: CustomLobbyWeaponData[];

    /** ArenaWeaponDatas */
    [1]: CustomLobbyWeaponData[];

    /** ArenaArmsRaceWeaponDatas */
    [2]: CustomLobbyArmsRaceWeaponData[];
}

File: Server/Data/Models/CustomLobby/DataForCustomLobbyCreation.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace TDS_Server.Data.Models.CustomLobby
{
    public class DataForCustomLobbyCreation
    {
#nullable disable
        [JsonProperty("0")]
        public List<CustomLobbyWeaponData> WeaponDatas { get; set; }

        [JsonProperty("1")]
        public List<CustomLobbyWeaponData> ArenaWeaponDatas { get; set; }

        [JsonProperty("2")]
        public List<CustomLobbyArmsRaceWeaponData> ArenaArmsRaceWeaponDatas { get; set; }
    }
}
